File: backend/models/Transaction.js
var mongoose = require('mongoose');


var transactionSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true
    },
    date: {
        type: String,
        required: true
    },
    reason: {
        type: String,
        required: true
    },
    amount: {
        type: Number,
        required: true
    },
    type: {
        type: String,
        required: true
    }
})


module.exports = mongoose.model('Transaction', transactionSchema);

File: app/lib/screens/TransactionPage/models/Transaction.dart
class Transaction {
  String name;
  String id;
  String date;
  String reason;
  double amount;
  String type;

  Transaction(
      this.id, this.amount, this.date, this.name, this.reason, this.type);

  Transaction.fromJson(Map<String, dynamic> json) {
    id = json['_id'];
    name = json['name'];
    date = json['date'];
    reason = json['reason'];
    amount = json['amount'].toDouble();
    type = json['type'];
  }
}
